src/core/thread/link_quality.hpp

```diff
@@ -214,23 +221,28 @@ class LinkQualityInfo
 /**
  * This function returns the current average noise floor level (in dBm).
  *
+ * @param[in]  aNoiseFloor  A reference to the noise state.
+ *
  * @returns The current average noise floor level (in dBm).
  */
-int8_t GetAverageNoiseFloor(void);
+int8_t GetAverageNoiseFloor(LinkQualityInfo &aNoiseFloor);
 
 /**
  * This method adds a new noise floor value (in dBm) to the running average.
  *
- * @param[in] aNoiseFloor    A new noise floor value (in dBm) to be added to the average.
+ * @param[in]  aNoiseFloor    A reference to the noise state.
+ * @param[in]  aNoise         A new noise floor value (in dBm) to be added to the average.
  *
  */
-void AddNoiseFloor(int8_t aNoiseFloor);
+void AddNoiseFloor(LinkQualityInfo &aNoiseFloor, int8_t aNoise);
 
 /**
  * This method clears the current average noise floor value.
  *
+ * @param[in]  aNoiseFloor  A reference to the noise state.
+ *
  */
-void ClearNoiseFloorAverage(void);
+void ClearNoiseFloorAverage(LinkQualityInfo &aNoiseFloor);
 
```

Comment:
You are correct that the noise floor estimator is not currently hooked in.  This is an open TODO.